### PR TITLE
Add new task to post-process marine DA

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
The work in this PR is only meant to bring us closer to a viable WCDA system. The refactoring of the marine DA to the new standard introduced by @aerorahul and used by @RussTreadon-NOAA and @CoryMartin-NOAA will be addressed after this [Epic](https://github.com/noaa-emc/gdasapp/issues/416) is resolved. 

### Motivation and context
This work adds a separate j-job ```JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT ``` that calls a script that will be in the GDASApp for the time being (PR to come once this is merged) and does the following:
- prepares the `SOCA` increment for `MOM6` IAU
- recursively apply the `SOCA2CICE` change of variable. A mapping from the 2D seaice analysis variable to the CICE6 dynamical and thermodynamic variables.
- merge the `Tref` increment from the `NSST` analysis with the `SOCA` increment   

### Summary of the change
- HPC environment: the new j-job runs a `JEDI` executable twice and one python script. All are serial jobs but the JEDI exec need to be called as an MPI job with 1 pe. 
- `jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT`, that script point to a ex-script that is not in the GDASApp develop yet.
- addition of the option to merge the Tref NSST increment with the MOM6 increment. This is triggered with the `DO_MERGENSST` switch
- The new j-job dependency was added, with the option to wait for the surface analysis file `sfcanl.nc` if `do_mergensst` is true.

This work is mentioned in the `global-workflow` issue #1480.
- Fixes issue https://github.com/NOAA-EMC/GDASApp/issues/418 

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
- This work was extracted from a branch that includes all the necessary changes to cycle a WCDA system. It was tested at c48o500 and c384o025.
- It will require a few PR's in the GDASApp to allow others to reproduce our results.
- Only tested on Hera
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks ok to me.

**Review by CoryMartin-NOAA — APPROVED** [review]

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+export STRICT="NO"
+source "${HOMEgfs}/ush/preamble.sh"
+WIPE_DATA="NO"
```
[the remark this answers is not in the document]
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT]: Just export it to shut shellcheck up.

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+export DATA="${DATAROOT}/${RUN}ocnanal_${cyc}"
+source "${HOMEgfs}/ush/jjob_header.sh" -e "ocnanalchkpt" -c "base ocnanal ocnanalchkpt"
+
+
+##############################################
+# Begin JOB SPECIFIC work
+##############################################
+
+export GDATE=$(date +%Y%m%d%H -d "${CDATE:0:8} ${CDATE:8:2} - ${assim_freq} hours")
```
[the remark this answers is not in the document]
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT]:
```suggestion
# Ignore possible spelling error (nothing is misspelled)
# shellcheck disable=SC2153
GDATE=$(date +%Y%m%d%H -d "${PDY} ${cyc} - ${assim_freq} hours")
export GDATE
```

File: jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export gPDY=${GDATE:0:8}
+export gcyc=${GDATE:8:2}
+export GDUMP=${GDUMP:-"gdas"}
+
+export GPREFIX="${GDUMP}.t${gcyc}z."
+export APREFIX="${CDUMP}.t${cyc}z."
```
[the remark this answers is not in the document]
Comment (WalterKolczynski-NOAA) [reply jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_CHKPT:24]:
```suggestion
# Ignore possible spelling error (nothing is misspelled)
# shellcheck disable=SC2153
export APREFIX="${CDUMP}.t${cyc}z."
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Just some shellcheck solutions. Want to get this in before the COM refactor breaks UFSDA.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres I took ashot as changing the PR title on my own but you seem to have reverted it back. Would you please give it a new title that is (a) shorter and (b) uses imperative mood (starts with a verb; it should complete the sentence "This PR will _______").

After that is done, I will merge.

**Comment by guillaumevernieres** [comment]
You're going to block a merge because you don't like the title of my PR @WalterKolczynski-NOAA ? I will revert to your title if that's what it takes, but do you have a valid reason for editing other people's PR title?

**Comment by guillaumevernieres** [comment]
Thanks all for your reviews. And whatever for the title ...

**Comment by WalterKolczynski-NOAA** [comment]
Thanks @guillaumevernieres

**Pull request merged**